### PR TITLE
WIP: scalable k-means

### PR DESCRIPTION
**Pull request by TomAugspurger**
Implements [`k-means||`](https://arxiv.org/abs/1203.6402), a scalable k-means initialization alternative to K-Means++

### CODE REVIEW

File: benchmarks/k_means.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+"""
+import argparse
+import logging
+import sys
+try:
+    import coloredlogs
+except ImportError:
+    pass
+else:
+    coloredlogs.install()
```
Comment (mrocklin) [inline benchmarks/k_means.py]:
```
mrocklin@carbon:~/workspace/dask-ml$ python benchmarks/k_means_kdd.py 
Traceback (most recent call last):
  File "benchmarks/k_means_kdd.py", line 12, in <module>
    import coloredlogs
ModuleNotFoundError: No module named 'coloredlogs'
```

File: benchmarks/k_means_kdd.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        logger.info("Using distributed mode")
+        client = Client(args.scheduler_address)
+        logger.info(client.scheduler_info())
+        fs = s3fs.S3FileSystem()
+        if fs.exists("dask-data"):
+            logger.info("Using cached dataset")
+        else:
+            logger.info("Uploading to cloud storage")
+            upload(local, fs)
+        path = "dask-data/kddcup/kdd.parq/"
```
Comment (mrocklin) [inline benchmarks/k_means_kdd.py]: I tried running this with a dask cluster on localhost and ran into errors here

Comment (TomAugspurger) [reply benchmarks/k_means_kdd.py]: Whoops, missing the `s3://`.

Some context, the kdd-cup dataset is the largest dataset used in the k-means|| paper, but it isn't that big... I can cluster the entire dataset just fine on my laptop.

I just setup a cluster to benchmark k-means on the airlines dataset.

Comment (mrocklin) [reply benchmarks/k_means_kdd.py]: We shouldn't necessarily need to engage s3 here though, no?  I wanted to run this locally just to see the diagnostic dashboard (things look great by the way) and ran into issues here

**Comment by TomAugspurger** [comment]
Oh, sorry mixing up my examples. Yes... that should be `path = local` I
think. I'll push an update.

File: benchmarks/k_means_kdd.py (hunk 2 above)
Comment (TomAugspurger) [reply benchmarks/k_means_kdd.py]: Oh I remember now. I assumed that using the distributed scheduler implied a remote cluster. I'll have to refactor it a bit more then.

**Comment by TomAugspurger** [comment]
For some reason, travis is pickup up https://github.com/dask/dask-ml/commit/a2541e4cc775f27e9cef9e5c8cb4378f5ac821bd as the merge commit, which is for #15 . Opening a new PR.

**Pull request closed without merging**